lib/tst_safe_macros.c

```diff
@@ -625,3 +625,14 @@ void safe_print_file(const char *file, const int lineno, char *path)
 
 	safe_fclose(file, lineno, NULL, pfile);
 }
+
+int safe_sscanf(const char *file, const int lineno, const char *restrict buffer, const char *restrict format, ...)
+{
+	va_list args;
+	va_start(args, format);
+	int ret = vsscanf(buffer, format, args);
+	va_end(args);
+	if (ret == EOF)
```

Comment:
This does not actually fail if we have less than expected number of conversion but some successful. You have to count the conversions and make sure that we got exactly the right number. We do have `tst_count_scanf_conversions()` function for exactly that.

Reply:
Thanks for the hint; from first instructions it wasn't clear you wanted to have that kind of check in place. Makes sense 👍🏻, will do ASAP

Reply:
I would first handle `EOF` as the other check is IMHO more specific.

Also, does it make sense to print `errno` (with `TERRNO`)? Man states:

       EILSEQ Input byte sequence does not form a valid character.

       EINVAL Not enough arguments; or format is NULL.

       ENOMEM Out of memory.